src/graph/planner/plan/PlanNode.cpp

```diff
@@ -334,11 +334,12 @@ void PlanNode::calcCost() {
   VLOG(1) << "unimplemented cost calculation.";
 }
 
-void PlanNode::setOutputVar(const std::string& var) {
+void PlanNode::setOutputVar(const std::string& var, const std::vector<std::string>& colNames) {
   auto* outputVarPtr = qctx_->symTable()->getVar(var);
   DCHECK(outputVarPtr != nullptr);
   auto oldVar = outputVar_->name;
   outputVar_ = outputVarPtr;
+  setColNames(colNames);
```

Comment:
could use the column names of variables as the colNames when outputVar_ is DataSet value type, such as:

```c++
if (colNames.empty() && !outputVar_->colNames.empty()) {
  setColNames(outputVar_->colNames);
}
```

Reply:
Sounds good. Fixed.